### PR TITLE
Added: code to get active xcode path

### PR DESCRIPTION
**Pull request by equinox2k**
I noticed ShaderGen assumed path of Xcode which caused problems on my machine as my Xcode's are installed under Applications as Xcode94.app, Xcode101.app etc

### CODE REVIEW

**Comment by equinox2k** [comment]
I see AppVeyor failed but dont think from what ive done as tested locally

File: src/ShaderGen.App/Program.cs [hunk 1]
```diff
@@ -519,6 +522,26 @@ private static string FindFxcExe()
...
+                {
+                    FileName = "xcode-select", // Specify exe name.
+                    Arguments = "--print-path",
+                    RedirectStandardOutput = true
+                };
+                using (Process process = Process.Start(startInfo))
+                {
+                    using (StreamReader reader = process.StandardOutput)
+                    {
+                        s_xcodePath = reader.ReadLine();
```
Comment (mellinoe) [inline src/ShaderGen.App/Program.cs]: The logic before was resilient to xcode not being installed. I believe this is going to throw an exception now if you don't have it installed.

**Comment by mellinoe** [comment]
This path actually isn't correct on my machine, and I think there may be a better way to find these applications. There is an `xcrun` utility which has a `--find` option to help locate things like this, apparently. On my machine, I get this output from the command:

```
xcrun --find metal
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/metal
xcrun --find metallib
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/metallib

xcrun --sdk iphoneos --find metal
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/metal
xcrun --sdk iphoneos --find metallib
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/metallib
```

The last two are the same for me, but I suppose they could be different, depending on the SDK (?)

File: src/ShaderGen.App/Program.cs (hunk 1 above)
Comment (equinox2k) [reply src/ShaderGen.App/Program.cs]: My assumption was if Mac OS should have Xcode :), will improve as suggested :)

**Comment by equinox2k** [comment]
I thought Xcode-select was way to go, maybe In your case you not installed the command line tools. Will look further.

**Comment by equinox2k** [comment]
@mellinoe see what you think of my latest change :)

**Comment by mellinoe** [comment]
Unfortunately, the current version isn't working. I got past a couple of initial problems, but the logic to locate `metal` and `metallib` isn't correct, at least for my machine. I think the correct way to find the tools is to use `xcrun --find` as I mentioned above. I am going to try to spend a little bit of time implementing that.

**Comment by mellinoe** [comment]
I merged in some extra changes on top of this, thanks for working on this!

**Pull request merged**